barretenberg/cpp/pil/vm2/tx_context.pil

```diff
@@ -168,6 +168,30 @@ namespace tx;
         public_inputs.cols[1]
     };
 
+    // Read reverted from PI
+
+    // Whether this tx reverted
+    pol commit tx_reverted;
+    #[INIT_TX_REVERTED]
+    start_tx * (reverted - tx_reverted) = 0;
+    // If reverted == 1, tx_reverted == 1, otherwise it must be continuous (see #[TX_REVERTED_CONTINUITY])
+    // from the start of the tx (see #[INIT_TX_REVERTED]).
+    #[SET_TX_REVERTED]
+    reverted * (1 - tx_reverted) = 0;
```

Comment:
Could combine these if we prefer:

- If we cannot revert at `start_tx`, `tx_reverted` must start at 0, and could probably use the existing `#[TX_REVERTED_CONTINUITY]` to constrain init
- Both these relations set `tx_reverted = reverted`, so could be something like `(start_tx + reverted) * (reverted - tx_reverted) = 0`